kajiki/xml_template.py

```diff
@@ -115,6 +116,9 @@ def compile(self):
             registries of the compiler ``compile`` should
             never be called twice or might lead to unexpected results.
         """
+        if len(self.doc.childNodes) != 1:
+            raise XMLTemplateCompileError('more than one children in document',
```

Comment:
Instead of failing can't we move forward in childNodes until a non `xml.etree.elementTree.Comment` node is found? That should do the trick

Reply:
https://github.com/CastixGitHub/kajiki/commit/03f14f4201144e012410e6f6447207ce86a22ea0
This should be better than skipping comments, but I don't think it's a good idea to do so, as you won't have a main element

Reply:
Well handling multiple nodes is a big change as people might be relying on the fact that secondary nodes are ignored. So I'd delay it in a further major release if we want to actually apply it.

On the other side skipping comments seems to do no harm at all so it's a first step we can apply right now.